src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts

```diff
@@ -198,4 +205,66 @@ export class CourseManagementTabBarComponent implements OnInit, OnDestroy, After
         const courseManagementRegex = /course-management\/[0-9]+(\/edit)?$/;
         return courseManagementRegex.test(this.router.url);
     }
+
+    private setExistingSummaryEntries() {
+        const numberRepositories =
+            this.course?.exercises
+                ?.filter((exercise) => exercise.type === 'programming')
+                .map((exercise) => exercise?.numberOfParticipations ?? 0)
+                .reduce((a, b) => a + b, 0) ?? 0;
+
+        const numberOfExercisesPerType = new Map<ExerciseType, number>();
+        this.course?.exercises?.forEach((exercise) => {
+            if (exercise.type === undefined) {
+                return;
+            }
+            const oldValue = numberOfExercisesPerType.get(exercise.type) ?? 0;
+            numberOfExercisesPerType.set(exercise.type, oldValue + 1);
+        });
+
+        const numberExams = this.course?.numberOfExams ?? 0;
+        const numberLectures = this.course?.lectures?.length ?? 0;
+        const numberStudents = this.course?.numberOfStudents ?? 0;
+        const numberTutors = this.course?.numberOfTeachingAssistants ?? 0;
+        const numberEditors = this.course?.numberOfEditors ?? 0;
+        const numberInstructors = this.course?.numberOfInstructors ?? 0;
+        const isTestCourse = this.course?.testCourse;
+
+        this.courseSummary = {
+            ...this.courseSummary,
+            'artemisApp.course.delete.summary.numberRepositories': numberRepositories,
+            'artemisApp.course.delete.summary.numberProgrammingExercises': numberOfExercisesPerType.get(ExerciseType.PROGRAMMING) ?? 0,
+            'artemisApp.course.delete.summary.numberModelingExercises': numberOfExercisesPerType.get(ExerciseType.MODELING) ?? 0,
+            'artemisApp.course.delete.summary.numberTextExercises': numberOfExercisesPerType.get(ExerciseType.TEXT) ?? 0,
+            'artemisApp.course.delete.summary.numberFileUploadExercises': numberOfExercisesPerType.get(ExerciseType.FILE_UPLOAD) ?? 0,
+            'artemisApp.course.delete.summary.numberQuizExercises': numberOfExercisesPerType.get(ExerciseType.QUIZ) ?? 0,
+            'artemisApp.course.delete.summary.numberExams': numberExams,
+            'artemisApp.course.delete.summary.numberLectures': numberLectures,
+            'artemisApp.course.delete.summary.numberStudents': numberStudents,
+            'artemisApp.course.delete.summary.numberTutors': numberTutors,
+            'artemisApp.course.delete.summary.numberEditors': numberEditors,
+            'artemisApp.course.delete.summary.numberInstructors': numberInstructors,
+            'artemisApp.course.delete.summary.isTestCourse': isTestCourse,
+        };
+    }
+
+    private fetchAndSetCourseDeletionSummary() {
+        this.courseAdminService.getDeletionSummary(this.course?.id!).subscribe({
```

Comment:
The client style is complaining here:

![image](https://github.com/user-attachments/assets/5e544731-ca2a-4f54-8308-56013826b665)

Maybe surround the code with an `if(this.course?.id) { ... }`